bundles/org.openhab.binding.guntamatic/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,596 @@
+				<label>Set Program of Heat Circle 3</label>
+			</channel>
+			<channel id="controlHeatCircProgram4" typeId="controlHeatCircProgram">
+				<label>Set Program of Heat Circle 4</label>
+			</channel>
+			<channel id="controlHeatCircProgram5" typeId="controlHeatCircProgram">
+				<label>Set Program of Heat Circle 5</label>
+			</channel>
+			<channel id="controlHeatCircProgram6" typeId="controlHeatCircProgram">
+				<label>Set Program of Heat Circle 6</label>
+			</channel>
+			<channel id="controlHeatCircProgram7" typeId="controlHeatCircProgram">
+				<label>Set Program of Heat Circle 7</label>
+			</channel>
+			<channel id="controlHeatCircProgram8" typeId="controlHeatCircProgram">
+				<label>Set Program of Heat Circle 8</label>
+			</channel>
+			<channel id="controlWwHeat0" typeId="controlWwHeat">
+				<label>Trigger Warm Wather Circle 0</label>
+			</channel>
+			<channel id="controlWwHeat1" typeId="controlWwHeat">
+				<label>Trigger Warm Wather Circle 1</label>
+			</channel>
+			<channel id="controlWwHeat2" typeId="controlWwHeat">
+				<label>Trigger Warm Wather Circle 2</label>
+			</channel>
+			<channel id="controlExtraWwHeat0" typeId="controlExtraWwHeat">
+				<label>Trigger Extra Warm Wather Circle 0</label>
+			</channel>
+			<channel id="controlExtraWwHeat1" typeId="controlExtraWwHeat">
+				<label>Trigger Extra Warm Wather Circle 1</label>
+			</channel>
+			<channel id="controlExtraWwHeat2" typeId="controlExtraWwHeat">
+				<label>Trigger Extra Warm Wather Circle 2</label>
+			</channel>
+		</channels>
+		<properties>
+			<property name="boilerApproval">PK002</property>
+			<property name="program">PR001</property>
+			<property name="heatCircProgram">HKx01</property>
+			<property name="wwHeat">BKx06</property>
+			<property name="extraWwHeat">ZKx06</property>
+		</properties>
+		<config-description>
+			<parameter name="hostname" type="text" required="true">
+				<context>network-address</context>
+				<label>Hostname</label>
+				<description>Hostname or IP address of the Guntamatic Heating System</description>
+			</parameter>
+			<parameter name="key" type="text" required="false">
+				<context>key</context>
+				<label>Key</label>
+				<description>Optional, but required to read protected parameters and to control the Guntamatic Heating System.
+					The
+					key needs to be reqested from Guntamatic support.</description>
+			</parameter>
+			<parameter name="refreshInterval" type="integer" unit="s" min="1" required="false">
+				<label>Refresh Interval</label>
+				<description>Interval the Guntamatic Heating System is polled in seconds. Default: 60s</description>
+			</parameter>
+			<parameter name="encoding" type="text" required="false">
+				<label>Encoding</label>
+				<description>Code page used by the Guntamatic Heating System. Default: 'windows-1252'</description>
+			</parameter>
+		</config-description>
+	</thing-type>
+	<thing-type id="therm">
```

Comment:
Do I miss something or are your 6 thing types strictly identical ?

I am not sure you really need different thing types in this case.

In case you keep 6 thing types, you could at least factorize the configuration part, by moving it in a file config.xml.

Reply:
> Do I miss something or are your 6 thing types strictly identical except thing label/description ?

Label/description (and validation status of the different things), but also some thing properties are different.

> You could have one unique thing type with the heat system type as configuration parameter.
> The technical parameters of each heat system type could be part of an enum class defining each heat system in your Java code.

I took the design decission to use different things instead of just a configuration parameter.

> In case you keep 6 thing types, you could at least factorize the configuration part, by moving it in a file config.xml.

I was already thinking about the usage of `config-description-ref`. I guess I will go for this improvement in order to avoid redundancies between the different things.

Reply:
> I took the design decission to use different things instead of just a configuration parameter.

That was probably a wrong design decision :) but ok